Sources/Build/BuildPlan.swift

```diff
@@ -773,12 +780,8 @@ public final class SwiftTargetBuildDescription {
         // FIXME: Eliminate side effect.
         result.append(try writeOutputFileMap().pathString)
 
-        switch target.type {
-        case .library, .test:
+        if target.type == .library || target.type == .test || (target.type == .executable && toolsVersion >= .v5_5 && self.needsToBeParsedAsLibrary) {
```

Comment:
another approach is to encapsulate all this into a `func needsToBeParsedAsLibrary()` which can have all this logic

```
func needsToBeParsedAsLibrary(target) -> Bool {
  switch target.type {
    case .library, .test:
      return true
    case .executable:
       return toolsVersion >= .v5_5 && self.needsToBeParsedAsLibrary
   default:
      ...
  }
}
```

Reply:
That's a good point.  Having this three times isn't ideal.  Will update.